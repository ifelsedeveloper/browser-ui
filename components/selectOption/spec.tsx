/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import SelectOption from '../selectOption'

describe('selectOption tests', () => {
  const baseComponent = (props?: object) => (
      <SelectOption id='selectOption' value='test' onChange={() => {}} {...props}>
        <option>some option</option>
      </SelectOption>
  )

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const tree = create(baseComponent()).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent({id: 'selectOptionId'}))
      const assertion = wrapper.find('#selectOptionId').length
      expect(assertion).toBe(1)
    })
  })

  describe('component behavior', () => {
    it('can have a custom title', () => {
      const wrapper = shallow(baseComponent({titleName: 'custom title here'}))
      const assertion = wrapper.html().includes('custom title here')
      expect(assertion).toBe(true)
    })

    it('can have an id', () => {
      const wrapper = shallow(baseComponent({id: 'selectOptionId'}))
      const assertion = wrapper.find('#selectOptionId').props().id
      expect(assertion).toEqual('selectOptionId')
    })

    it('can be set as multiple', () => {
      const wrapper = shallow(baseComponent({multiple: true}))
      const assertion = wrapper.find('#selectOption').props().multiple
      expect(assertion).toBe(true)
    })

    it('can be set autoFocus', () => {
      const wrapper = shallow(baseComponent({autoFocus: true}))
      const assertion = wrapper.find('#selectOption').props().autoFocus
      expect(assertion).toEqual(true)
    })

    it('can be disabled', () => {
      const wrapper = shallow(baseComponent({disabled: true}))
      const assertion = wrapper.find('#selectOption').props().disabled
      expect(assertion).toBe(true)
    })

    it('can respond to change', () => {
      const value = 'FANCY SELECT CHANGED'
      const onChange = jest.fn()
      const wrapper = shallow(baseComponent({onChange}))
      wrapper.find('#selectOption').simulate('change', value)
      expect(onChange).toBeCalledWith(value)
    })
  })
})
