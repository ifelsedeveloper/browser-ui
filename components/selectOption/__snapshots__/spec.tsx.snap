// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectOption tests basic tests matches the snapshot 1`] = `
.c2 {
  box-sizing: border-box;
  position: relative;
  border-radius: 4px;
  box-shadow: 0px 2px 8px -4px rgba(0,0,0,1);
  display: block;
  color: rgb(68,68,68);
  font-size: 13px;
  font-family: inherit;
  border: solid 1px rgba(0,0,0,0.2);
  outline: none;
  padding: 6px 12px 6px 6px;
  width: 100%;
  background-color: #fbfbfb;
  background-size: 12px 12px;
  -webkit-appearance: none;
  min-height: 32px;
  margin-bottom: 12px;
}

.c1 {
  position: relative;
}

.c1::after {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 1;
  content: "â–¼";
  position: absolute;
  top: 0;
  right: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-height: 100%;
  height: 100%;
  padding: 0 8px;
  font-size: 13px;
}

.c0 {
  margin: 0 0 4px 2px;
}

<div>
  <p
    className="c0"
  />
  <div
    className="c1"
    disabled={false}
    multiple={false}
  >
    <select
      autoFocus={false}
      className="c2"
      disabled={false}
      id="selectOption"
      multiple={false}
      onChange={[Function]}
      value="test"
    >
      <option>
        some option
      </option>
    </select>
  </div>
</div>
`;
